backend/src/main/java/cps/controllers/SignupController.java

```diff
@@ -1,19 +1,38 @@
 package cps.controllers;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
 
 import cps.models.RequestBodies.SignupRequestbody;
 import cps.models.User;
 
+@RestController
 public class SignupController {
 
-  public void signup(@RequestBody SignupRequestbody signupRequest) {
-    String firstname = signupRequest.getFirstname();
-    String lastname = signupRequest.getLastname();
-    String email = signupRequest.getEmail();
-    String password = signupRequest.getPassword();
-    User newUser = new User(firstname, lastname, email, password);
-    
+  @PostMapping("/signup")
+  public ResponseEntity<String> signup(@RequestBody SignupRequestbody signupRequest) {
+    try {
+      // Extract details from the request
+      String firstname = signupRequest.getFirstname();
+      String lastname = signupRequest.getLastname();
+      String email = signupRequest.getEmail();
+      String password = signupRequest.getPassword();
 
+      // Create a new User object
+      User newUser = new User(firstname, lastname, email, password);
+
+      // Save the user to the database
+      newUser.save();
+
+      // Return a success message
+      return new ResponseEntity<>("User registered successfully", HttpStatus.CREATED);
```

Comment:
I would make a response object with a message:"success message" just to make it return a json in the request rather than plain text.

Reply:
refer to this:
https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci80MWIzM2E3YWUyNjYzNGUzNGI5YzYxMGZmYWRhMDZjMDdiNDU2M2M5L2YvYmFja2VuZC9zcmMvbWFpbi9qYXZhL2Nwcy9jb250cm9sbGVycy9Mb2dpbkNvbnRyb2xsZXIuamF2YT91cmw9aHR0cHMlM0ElMkYlMkZnaXRodWIuY29tJTJGRnJhbmNvRG9taW5ndWV6JTJGU09FTjM0My1DUFMuZ2l0JmxpbmVzPTMwLTMy?origin=gitlens